src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -38,11 +41,145 @@ internal ILGeneratorImpl(MethodBuilder methodBuilder, int size)
 
         public override int ILOffset => _il.Offset;
 
-        public override void BeginCatchBlock(Type? exceptionType) => throw new NotImplementedException();
-        public override void BeginExceptFilterBlock() => throw new NotImplementedException();
-        public override Label BeginExceptionBlock() => throw new NotImplementedException();
-        public override void BeginFaultBlock() => throw new NotImplementedException();
-        public override void BeginFinallyBlock() => throw new NotImplementedException();
+        public override void BeginCatchBlock(Type? exceptionType)
+        {
+            if (_currentExStackIndex <= -1)
```

Comment:
Is this expression important as is? As in, can this be changed to `_currentExStackIndex < 0`?

Reply:
Instead of comparing to a specific index, would it be more appropriate to compare against the length of `_exceptionStack`?

Reply:
> Instead of comparing to a specific index, would it be more appropriate to compare against the length of _exceptionStack?

Thought if will be clearer, than using `_exceptionStack.length`, though have no strong opinion.

Reply:
Ah, actually it seems you meant to compare the `_currentExStackIndex` with `_exceptionStack` length.